docs/source/java/cdata.rst

```diff
@@ -0,0 +1,313 @@
+            <maven.compiler.target>8</maven.compiler.target>
+        </properties>
+        <dependencies>
+            <dependency>
+                <groupId>org.bytedeco</groupId>
+                <artifactId>javacpp</artifactId>
+                <version>1.5.7</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.arrow</groupId>
+                <artifactId>arrow-c-data</artifactId>
+                <version>7.0.0</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.arrow</groupId>
+                <artifactId>arrow-memory-netty</artifactId>
+                <version>7.0.0</version>
+            </dependency>
+        </dependencies>
+    </project>
+
+.. code-block:: java
+
+    import org.bytedeco.javacpp.annotation.Platform;
+    import org.bytedeco.javacpp.annotation.Properties;
+    import org.bytedeco.javacpp.tools.InfoMap;
+    import org.bytedeco.javacpp.tools.InfoMapper;
+
+    @Properties(
+            target = "CDataInterfaceLibrary",
+            value = @Platform(
+                    include = {"CDataInterfaceLibrary.h"},
+                    compiler = {"cpp11"},
+                    linkpath = {"/arrow/cpp/build/debug/"},
+                    link = {"arrow"}
+            )
+    )
+    public class CDataInterfaceLibraryConfig implements InfoMapper {
+        @Override
+        public void map(InfoMap infoMap) {
+        }
+    }
+
+.. code-block:: shell
+
+    // Compile our Java code
+    javac -cp javacpp-1.5.7.jar CDataInterfaceLibraryConfig.java
+
+    // Generate CDataInterfaceLibrary
+    java -jar javacpp-1.5.7.jar CDataInterfaceLibraryConfig.java
+
+    // Generate libjniCDataInterfaceLibrary.dylib
+    java -jar javacpp-1.5.7.jar CDataInterfaceLibrary.java
+
+    // Validate libjniCDataInterfaceLibrary.dylib created
+    otool -L macosx-x86_64/libjniCDataInterfaceLibrary.dylib
+    macosx-x86_64/libjniCDataInterfaceLibrary.dylib:
+            libjniCDataInterfaceLibrary.dylib (compatibility version 0.0.0, current version 0.0.0)
+            @rpath/libarrow.800.dylib (compatibility version 800.0.0, current version 800.0.0)
+            /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1200.3.0)
+            /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
+
+**Java Test**
+
+Let's create a Java class to Test C Data Interface from Java to C++:
+
+.. code-block:: java
+
+    import org.apache.arrow.c.ArrowArray;
+    import org.apache.arrow.c.ArrowSchema;
+    import org.apache.arrow.c.Data;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.BigIntVector;
+
+    public class TestCDataInterfaceV5 {
+
+        public static void main(String[] args) {
+            CDataInterfaceLibrary.ArrowSchema arrowSchema = new CDataInterfaceLibrary.ArrowSchema();
```

Comment:
Can we pick different names for these variables? It gets very confusing to have lots of things with the same name.

Reply:
Changed